packages/x-data-grid/src/utils/css/themeManager.ts

```diff
@@ -0,0 +1,58 @@
+import { useTheme, type Theme } from '@mui/material/styles';
+import { stylesToString } from './stylesToString';
+import { transformMaterialUITheme } from './transformVariables';
+
+const STYLE_ID = 'mui-x-data-grid';
+const CLASSNAME_PREFIX = 'MuiDataGridVariables';
+
+let element = undefined as HTMLStyleElement | undefined;
+
+let nextClassNameId = 1;
+const classNameByTheme = new WeakMap<object, string>();
+
+export function useThemeVariablesClassName() {
+  return getClassNameForTheme(useTheme());
+}
+
+function getClassNameForTheme(theme: Theme) {
+  let className = classNameByTheme.get(theme);
+  if (className) {
+    return className;
+  }
+
+  className = `${CLASSNAME_PREFIX}-${nextClassNameId}`;
+  nextClassNameId += 1;
+
+  classNameByTheme.set(theme, className);
+
+  injectStyles(`.${className}`, transformMaterialUITheme(theme));
```

Comment:
Eventually, the transformer function will be passed to the unstyled data grid, right?

Reply:
Yes, I just wanted to decouple the functions for now, but that will eventually live in the `/material` folder along with anything that imports from `@mui/material/styles`.